pkg/qtv/protocol_reader.go

```diff
@@ -376,29 +376,48 @@ func (qp *qProtocol) readPlayerInfo() error {
 	return nil
 }
 
-func (qp *qProtocol) readEntityNum() (entNum uint, flags uint) {
+func (qp *qProtocol) readEntityNum() (entNum uint, flags uint, moreflags uint) {
 	flags = uint(qp.r.GetUint16())
 
 	if flags == 0 {
-		return 0, 0
+		return 0, 0, 0
 	}
 
 	entNum = flags & 511
 	flags &^= 511
 
 	if (flags & uMoreBits) != 0 {
 		flags |= uint(qp.r.GetByte())
+		if (flags & uFteEvenMore) != 0 {
+			moreflags |= uint(qp.r.GetByte())
+			if (moreflags & uFteEvenMore) != 0 {
+				moreflags |= uint(qp.r.GetByte()) << 8
+			}
+
+			if (moreflags & uFteEntityDbl) != 0 {
+				entNum += 512
+			}
+			if (moreflags & uFteEntityDbl2) != 0 {
+				entNum += 1024
+			}
+		}
 	}
 
-	return entNum, flags
+	return entNum, flags, moreflags
 }
 
-func (qp *qProtocol) readEntityDelta(old *entityState, new *entityState, flags uint, forcereLink bool) {
+func (qp *qProtocol) readEntityDelta(old *entityState, new *entityState, flags uint, moreflags uint, forcereLink bool) {
 	*new = *old
 
 	if (flags & uModel) != 0 {
-		new.modelIndex = qp.r.GetByte()
+		new.modelIndex = uint16(qp.r.GetByte())
+		if (moreflags & uFteModelDbl) != 0 {
+			new.modelIndex += 256
+		}
+	} else if (moreflags & uFteModelDbl) != 0 {
+		new.modelIndex = qp.r.GetUint16()
```

Comment:
It's a bug in mvdsv/ezquake that this fully extended model index behavior is not implemented, will fix that later in both.